File: timecapsule-database/src/main/java/com/timecapsule/infra/converter/StringListConverter.java
package com.timecapsule.infra.converter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
public class StringListConverter implements AttributeConverter<List<String>, String> {

    @Override
    public String convertToDatabaseColumn(List<String> list) {
        return String.join(",", list); // => <= >=
    }

    @Override
    public List<String> convertToEntityAttribute(String dbData) {
        return new ArrayList<>(Arrays.asList(dbData.split(",")));
    }
}

File: common/timecapsule-database/src/main/kotlin/com/timecapsule/database/converter/StringListConverter.kt
package com.timecapsule.database.converter

import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter
class StringListConverter : AttributeConverter<List<String>, String> {
    override fun convertToDatabaseColumn(list: List<String>): String = list.joinToString(",")

    override fun convertToEntityAttribute(dbData: String): List<String> = dbData.split(",")
}
